tests/cpp/test_alg_simp.cpp

```diff
@@ -32,4 +33,72 @@ TI_TEST("simplify_add_zero") {
   // .. more tests, assuming instruction order not shuffled
 }
 
+TI_TEST("simplify_multiply_one") {
+  auto block = std::make_unique<Block>();
+
+  auto global_load_addr =
+      block->push_back<GlobalTemporaryStmt>(0, VectorType(1, DataType::f32));
+  auto global_load = block->push_back<GlobalLoadStmt>(global_load_addr);
+  auto one = block->push_back<ConstStmt>(TypedConstant(1.0f));
+  auto mul1 =
+      block->push_back<BinaryOpStmt>(BinaryOpType::mul, one, global_load);
+  auto mul2 = block->push_back<BinaryOpStmt>(BinaryOpType::mul, mul1, one);
+  auto zero = block->push_back<ConstStmt>(TypedConstant(0.0f));
+  auto div = block->push_back<BinaryOpStmt>(BinaryOpType::div, zero, one);
+  auto sub = block->push_back<BinaryOpStmt>(BinaryOpType::sub, mul2, div);
+  auto global_store_addr =
+      block->push_back<GlobalTemporaryStmt>(4, VectorType(1, DataType::f32));
+  auto global_store = block->push_back<GlobalStoreStmt>(global_store_addr, sub);
+
+  irpass::typecheck(block.get());
+  TI_CHECK(block->size() == 10);
+
+  // irpass::print(block.get());
+
+  irpass::alg_simp(block.get(), CompileConfig());  // should eliminate mul, div, sub
+  irpass::die(block.get());       // should eliminate zero, one
+
+  // irpass::print(block.get());
+
+  TI_CHECK(block->size() == 4);  // two addresses, one load, one store
+  TI_CHECK((*block)[0]->is<GlobalTemporaryStmt>());
+}
+
+TI_TEST("simplify_multiply_zero_fast_math") {
+  auto block = std::make_unique<Block>();
+
+  auto global_load_addr =
+      block->push_back<GlobalTemporaryStmt>(0, VectorType(1, DataType::i32));
+  auto global_load = block->push_back<GlobalLoadStmt>(global_load_addr);
+  auto zero = block->push_back<ConstStmt>(TypedConstant(0));
+  auto mul =
+      block->push_back<BinaryOpStmt>(BinaryOpType::mul, global_load, zero);
+  auto one = block->push_back<ConstStmt>(TypedConstant(1));
+  auto add = block->push_back<BinaryOpStmt>(BinaryOpType::add, mul, one);
+  auto global_store_addr =
+      block->push_back<GlobalTemporaryStmt>(4, VectorType(1, DataType::i32));
+  auto global_store = block->push_back<GlobalStoreStmt>(global_store_addr, add);
+
+  irpass::typecheck(block.get());
+  TI_CHECK(block->size() == 8);
+
+  // irpass::print(block.get());
+
+  CompileConfig config_without_fast_math;
+  config_without_fast_math.fast_math = false;
```

Comment:
I like your treatment here: both `fast_math` on/off situations are tested. This means good code coverage. Great job!